ui/admin/app/components/form/field/list-wrapper/key-value/index.hbs

```diff
@@ -2,40 +2,76 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: BUSL-1.1
 }}
-{{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
 
+{{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
   <Hds::Table class='list-wrapper-field' name={{@name}}>
     <:head as |H|>
       <H.Tr>
-        <H.Th id={{keyHeaderID}}>{{t 'form.key.label'}}</H.Th>
-        <H.Th id={{valueHeaderID}}>{{t 'form.value.label'}}</H.Th>
+        <H.Th id={{keyHeaderID}}>
+          {{#if @keyLabel}}
+            {{@keyLabel}}
+          {{else}}
+            {{t 'form.key.label'}}
+          {{/if}}</H.Th>
+        <H.Th id={{valueHeaderID}}>
+          {{#if @valueLabel}}
+            {{@valueLabel}}
+          {{else}}
+            {{t 'form.value.label'}}
+          {{/if}}
+        </H.Th>
         <H.Th>
           {{t 'titles.actions'}}
-
         </H.Th>
       </H.Tr>
     </:head>
     <:body as |B|>
       {{#each this.options as |option index|}}
         <B.Tr>
           <B.Td>
-            <Hds::Form::TextInput::Field
-              @value={{option.key}}
-              @type='text'
-              disabled={{@disabled}}
-              aria-labelledby={{keyHeaderID}}
-              {{on 'input' (set-from-event option 'key')}}
-            />
+            {{yield
+              (hash
+                text=(component
+                  'form/field/list-wrapper/key-value/text'
+                  disabled=@disabled
+                  value=option.key
+                  ariaLabel=keyHeaderID
```

Comment:
I don't think the `ariaLabel`'s can be this generated GUID, it should be human readable otherwise the screen reader would be reading a GUID out haha

Do we not have a guaranteed label anymore? Should it have still been the `aria-labelledby`?

Reply:
oops yes, the other ones have `aria-labelledby`